plugins/modules/ec2_vol.py

```diff
@@ -457,6 +459,9 @@ def create_volume(module, ec2_conn, zone):
 
     volume = get_volume(module, ec2_conn)
 
+    if module.check_mode:
+        module.exit_json(changed=True, msg='Would have created a volume if not in check mode.')
+
     if volume is None:
```

Comment:
One minor niggle for the future (not worth blocking this over).  I find it's less likely to lead to weird edge cases if you make your check_mode tests as late as possible.  Here for example, if someone re-wrote the flow of main() and you'd sometimes be passed a volume, performing the check_mode test before you test volume would result in broken idempotency.